File: Models/Menus/MenuActivityPermissionModel.cs
using System.ComponentModel.DataAnnotations.Schema;
using Berger.Data.MsfaEntity.Menus;
using Berger.Data.MsfaEntity.Users;

namespace BergerMsfaApi.Models.Menus
{
    public class MenuActivityPermissionModel
    {
        public int Id { get; set; }
        public int RoleId { get; set; }

        [ForeignKey("RoleId")]
        public Role Role { get; set; }

        public int ActivityId { get; set; }

        [ForeignKey("ActivityId")]
        public MenuActivity Activity { get; set; }


        public bool CanView { get; set; }
        public bool CanUpdate { get; set; }
        public bool CanInsert { get; set; }
        public bool CanDelete { get; set; }
    }
}

File: ClientApp/src/app/Shared/Entity/Menu/menu-activity-permission.model.ts
export class MenuActivityPermission {
    public id: number;
    public status: number;
    public roleId: number;
    public activityId: number;
    public canView: boolean=false;
    public canUpdate: boolean=false;
    public canInsert: boolean=false;
    public canDelete: boolean=false;
}